hook-worker/src/worker.rs

```diff
@@ -165,65 +172,87 @@ impl<'p> WebhookWorker<'p> {
         if transactional {
```

Comment:
Note that the duplicated code for txn vs non-txn predated this PR. Refactoring to not have any duplication would be a lot of work, I think, and maybe/hopefully we'll drop one of these modes soon anyway?

Reply:
Yup, my understanding is that we want to keep our options open while we're evaluating the performance profile of the workload, but we should drop one of them before the end of Q1. One of them is basically dead code I would not feel comfortable switching to anyway.